keyboards/keebio/bdn9/proton_c/eeconfig.h

```diff
@@ -0,0 +1,105 @@
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef EECONFIG_H
+#define EECONFIG_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+
+#define EECONFIG_MAGIC_NUMBER                       (uint16_t)0xFEED
+#define EECONFIG_MAGIC_NUMBER_OFF                   (uint16_t)0xFFFF
+
+/* EEPROM parameter address */
+#define EECONFIG_MAGIC                              (uint16_t *)0
+#define EECONFIG_DEBUG                               (uint8_t *)2
+#define EECONFIG_DEFAULT_LAYER                       (uint8_t *)3
+#define EECONFIG_KEYMAP                              (uint8_t *)4
+#define EECONFIG_MOUSEKEY_ACCEL                      (uint8_t *)5
+#define EECONFIG_BACKLIGHT                           (uint8_t *)6
+#define EECONFIG_AUDIO                               (uint8_t *)7
+#define EECONFIG_RGBLIGHT                           (uint32_t *)8
+#define EECONFIG_UNICODEMODE                        (uint8_t *)12
+#define EECONFIG_STENOMODE                          (uint8_t *)13
+// EEHANDS for two handed boards
+#define EECONFIG_HANDEDNESS                         (uint8_t *)14
+#define EECONFIG_KEYBOARD                          (uint32_t *)15
+#define EECONFIG_USER                              (uint32_t *)19
+#define EECONFIG_VELOCIKEY                          (uint8_t *)23
+
+#define EECONFIG_HAPTIC                            (uint32_t *)24
+#define EECONFIG_RGB_MATRIX                        (uint32_t *)28
+#define EECONFIG_RGB_MATRIX_SPEED                   (uint8_t *)32
+
+/* debug bit */
+#define EECONFIG_DEBUG_ENABLE                       (1<<0)
+#define EECONFIG_DEBUG_MATRIX                       (1<<1)
+#define EECONFIG_DEBUG_KEYBOARD                     (1<<2)
+#define EECONFIG_DEBUG_MOUSE                        (1<<3)
+
+/* keyconf bit */
+#define EECONFIG_KEYMAP_SWAP_CONTROL_CAPSLOCK       (1<<0)
+#define EECONFIG_KEYMAP_CAPSLOCK_TO_CONTROL         (1<<1)
+#define EECONFIG_KEYMAP_SWAP_LALT_LGUI              (1<<2)
+#define EECONFIG_KEYMAP_SWAP_RALT_RGUI              (1<<3)
+#define EECONFIG_KEYMAP_NO_GUI                      (1<<4)
+#define EECONFIG_KEYMAP_SWAP_GRAVE_ESC              (1<<5)
+#define EECONFIG_KEYMAP_SWAP_BACKSLASH_BACKSPACE    (1<<6)
+#define EECONFIG_KEYMAP_NKRO                        (1<<7)
+
+
+bool eeconfig_is_enabled(void);
+bool eeconfig_is_disabled(void);
+
+void eeconfig_init(void);
+void eeconfig_init_quantum(void);
+void eeconfig_init_kb(void);
+void eeconfig_init_user(void);
+
+void eeconfig_enable(void);
+
+void eeconfig_disable(void);
+
+uint8_t eeconfig_read_debug(void);
+void eeconfig_update_debug(uint8_t val);
+
+uint8_t eeconfig_read_default_layer(void);
+void eeconfig_update_default_layer(uint8_t val);
+
+uint8_t eeconfig_read_keymap(void);
+void eeconfig_update_keymap(uint8_t val);
+
+uint8_t eeconfig_read_backlight(void);
```

Comment:
Good point about the ifdef locking these out - theoretically these shouldn't get compiled in anyway if the backlight feature is disabled... and in fact the ifdef is not present in the .c file.

Reply:
I think if you turn on `BACKLIGHT_ENABLE` and add `BACKLIGHT_CUSTOM_DRIVER = yes` in rules.mk, this file is not needed.

Reply:
oh nice. thanks for the tip.

Reply:
the problem there is that it includes tmk's backlight.h which has collisions with this implementation.

Reply:
You shouldn't need to reimplement the functions in there, only the actual implementation from quantum.c - backlight.h just manages the EEPROM and backlight struct, really.